src/Hooks/useVonageSinglePublisher.js

```diff
@@ -10,28 +10,37 @@ export const useVonageSinglePublisher = (session, hostName, publisherName) => {
   const navigate = useNavigate();
   const { dispatch } = useContext(UserContext);
   const [publishers, setPublishers] = useState();
+  const languagePublishing = {};
+  const localPublishers = {};
+  let publisherID = 'language-channels-container';
 
-  const createPublisher = () => {
-    const localPublishers = {};
-    let publisherID = 'language-channels-container';
+  const recordLangPublisher = (languageCode) => {
+    if (!Object.keys(languagePublishing).includes(languageCode)) return;
+    languagePublishing[languageCode] = true;
+    if (Object.values(languagePublishing).every((item) => item)) {
+      // Publish Video
+      createTokboxPublisher(hostName, publisherName, true, session, recordLangPublisher).then((publisher) => {
+        localPublishers[publisherName] = publisher;
+      });
+    }
+  };
 
+  const createPublisher = () => {
+    targetLanguagesKeys().forEach((langCode) => {
+      languagePublishing[langCode] = false;
+    });
     // Publish Audio
     targetLanguagesKeys().forEach((langCode) => {
-      createTokboxPublisher(langCode, publisherID, false, session).then((publisher) => {
+      createTokboxPublisher(langCode, publisherID, false, session, recordLangPublisher).then((publisher) => {
         localPublishers[langCode] = publisher;
       });
     });
 
     // Publish FLOOR Audio
-    createTokboxPublisher('FLOOR', publisherID, false, session).then((publisher) => {
+    createTokboxPublisher('FLOOR', publisherID, false, session, recordLangPublisher).then((publisher) => {
       localPublishers['FLOOR'] = publisher;
     });
 
-    // Publish Video
-    createTokboxPublisher(hostName, publisherName, true, session).then((publisher) => {
-      localPublishers[publisherName] = publisher;
-    });
-
```

Comment:
let's just move this code before the language channel publishing and remove the other changes

Reply:
I don't think sending a callback function and storing `languagePublishing` is useful.

Reply:
I already tried this and it was failing as well making it Intermittent.

Reply:
@beinghaziq I don't see any use of `languagePublishing`, then why are we storing publishing info on this?